plugins/Marketplace/PackageModel.py

```diff
@@ -250,3 +277,148 @@ def isCompatibleMaterialStation(self) -> bool:
     @pyqtProperty(bool, constant = True)
     def isCompatibleAirManager(self) -> bool:
         return self._is_compatible_air_manager
+
+    # --- manage buttons signals ---
+
+    stateManageButtonChanged = pyqtSignal()
+
+    installPackageTriggered = pyqtSignal(str)
+
+    uninstallPackageTriggered = pyqtSignal(str)
+
+    updatePackageTriggered = pyqtSignal(str)
+
+    enablePackageTriggered = pyqtSignal(str)
+
+    disablePackageTriggered = pyqtSignal(str)
+
+    recentlyInstalledChanged = pyqtSignal(bool)
+
+    # --- enabling ---
+
+    @pyqtProperty(str, notify = stateManageButtonChanged)
+    def stateManageEnableButton(self) -> str:
+        """The state of the manage Enable Button of this package"""
+        if self._is_enabling == ManageState.PROCESSING:
+            return "busy"
+        if self._package_type == "material" or not self._is_installed:
+            return "hidden"
+        if self._is_installed and self._is_active:
+            return "secondary"
+        return "primary"
+
+    @property
+    def is_enabling(self) -> ManageState:
```

Comment:
I don't think that you need this, but this is probably a consequence of having this handled in the model and not the view.